File: medium/zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1:
            return s
        rows = ['']*numRows
        reverse = False
        curr_row = 0
        forward = True
        for char in s:
            rows[curr_row] += char
            if curr_row == 0:
                forward = True
            if curr_row == len(rows)-1:
                forward = False
            if forward:
                curr_row+=1
            else:
                curr_row-=1
        word = ''
        for row in rows:
            word+=row
        return word

File: medium/zigzag-conversion.cpp
#include<bits/stdc++.h>;
using namespace std;

class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1){
            return s;
        }
        vector<string> words;
        int i = 0;
        while(i < numRows) {
            words.push_back("");
            i++;
        }
        int curr = 0;
        bool forward = true;
        for(int i = 0; i < s.size(); i++)
        {
            words[curr]+=s[i];
            if (curr == 0)
            {
                forward = true;
            }
            if (curr == numRows-1)
            {
                forward = false;
            }
            if (forward)
            {
                curr++;
            }
            else
            {
                curr--;
            }
        }
        string ans;
        for(int i = 0; i< numRows; i++)
        {
            ans+=words[i];
        }
        return ans;
    }
};